extensions/src/main/java/io/lionweb/repoclient/RepoSerialization.java

```diff
@@ -0,0 +1,261 @@
+package io.lionweb.repoclient;
+
+import static io.lionweb.lioncore.java.serialization.LowLevelJsonSerialization.groupNodesIntoSerializationBlock;
+
+import io.lionweb.lioncore.java.serialization.LowLevelJsonSerialization;
+import io.lionweb.lioncore.java.serialization.data.SerializedChunk;
+import io.lionweb.lioncore.java.serialization.data.SerializedClassifierInstance;
+import io.lionweb.repoclient.api.BulkAPIClient;
+import io.lionweb.repoclient.api.RawBulkAPIClient;
+import io.lionweb.serialization.extensions.AdditionalAPIClient;
+import io.lionweb.serialization.extensions.BulkImport;
+import io.lionweb.serialization.extensions.Compression;
+import io.lionweb.serialization.extensions.TransferFormat;
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import java.util.zip.ZipOutputStream;
+import kotlin.text.Charsets;
+
+/** This class contains the logic to store and retrieve entire repositories at once. */
+public class RepoSerialization {
+  private int nNodesThreshold = 100_000;
+  private TransferFormat transferFormat = TransferFormat.FLATBUFFERS;
+  private Compression compression = Compression.DISABLED;
+
+  /**
+   * Download all the content of the repository accessed by the apiClient into a directory. In this
+   * directory one file per partition is created. The file is in JSON format.
+   */
+  public <C extends RawBulkAPIClient & BulkAPIClient> void downloadRepoAsDirectory(
+      C apiClient, File directory) throws IOException {
+    for (String partitionID : apiClient.listPartitionsIDs()) {
+      String partitionData =
+          apiClient.rawRetrieve(Collections.singletonList(partitionID), Integer.MAX_VALUE);
+      File partitionFile = new File(directory, partitionID + ".json");
+      Files.write(
+          partitionFile.toPath(),
+          Collections.singletonList(partitionData),
+          StandardOpenOption.CREATE,
+          StandardOpenOption.TRUNCATE_EXISTING);
+    }
+  }
+
+  /**
+   * Download all the content of the repository accessed by the apiClient into a zip file. In this
+   * zip file, one entry per partition is created. The entry is in JSON format.
+   */
+  public <C extends RawBulkAPIClient & BulkAPIClient> void downloadRepoAsZip(
+      C apiClient, File zipFile) throws IOException {
+    try (FileOutputStream fos = new FileOutputStream(zipFile);
```

Comment:
I'm not sure about current best practices in Java, but it used to be recommended to have a BufferedStream somewhere in there.

Reply:
Yes, you are right. It should improve performance